test/pages/HomePage.ts

```diff
@@ -0,0 +1,30 @@
+import { Locator, Page } from "@playwright/test";
+import { home_url } from "../helper-config";
+
+export class HomePage {
+  page: Page;
+  loginLinkLocator: Locator;
+  signupLinkLocator: Locator;
+  welcomeLocator: Locator;
+
+  constructor({ page }: { page: Page }) {
+    this.page = page;
+    this.loginLinkLocator = page.getByRole("link", { name: "Log In" });
+    this.signupLinkLocator = page.getByRole("link", { name: "Sign Up" });
+    this.welcomeLocator = page.getByRole("heading", {
+      name: "Welcome to Open Library",
+    });
+  }
+
+  async navigateToHome() {
+    await this.page.goto(home_url);
+  }
+
+  async goToLogin() {
+    await this.loginLinkLocator.click();
```

Comment:
this will take you to the LoginPage, so after the click, we should
1. wait for the page to load
2. return a `LoginPage` object

same for other functions that traverse through pages